mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td

```diff
@@ -1369,12 +1369,12 @@ def Tosa_ReduceSumOp : Tosa_Op<"reduce_sum", [
   }];
 
   let arguments = (ins
-    Tosa_Tensor1Dto4D:$input,
+    Tosa_Tensor:$input,
     I64Attr:$axis
   );
 
   let results = (outs
-    Tosa_Tensor1Dto4D:$output
+    Tosa_Tensor:$output
   );
```

Comment:
Do you know why this limitation exists?
Is this not enforced by any verifiers?
Could you add a small test with more than four dimensions?